timemachine/cpp/src/chiral_atom_restraint.cu

```diff
@@ -0,0 +1,56 @@
+#include "chiral_atom_restraint.hpp"
+#include "gpu_utils.cuh"
+#include "k_chiral_restraint.cuh"
+#include "kernel_utils.cuh"
+#include "math_utils.cuh"
+#include <vector>
+
+namespace timemachine {
+
+template <typename RealType>
+ChiralAtomRestraint<RealType>::ChiralAtomRestraint(const std::vector<int> &idxs) : R_(idxs.size() / 4) {
+
+    if (idxs.size() % 4 != 0) {
+        throw std::runtime_error("idxs.size() must be exactly 4*k!");
+    }
+
+    gpuErrchk(cudaMalloc(&d_idxs_, R_ * 4 * sizeof(*d_idxs_)));
+    gpuErrchk(cudaMemcpy(d_idxs_, &idxs[0], R_ * 4 * sizeof(*d_idxs_), cudaMemcpyHostToDevice));
```

Comment:
nit: should we switch to `DeviceBuffer` for new code?

Reply:
Probably - this was copy and pasted from the `HarmonicBond` code, should we just try to do them all during deboggling week for consistency?